test_fixtures/gold_output_files/README.md

```diff
@@ -0,0 +1,6 @@
+## Updating golds
+
+To re-generate the gold files in this directory using the over_react builder, run:
+```dart
+dart tool/update_gold_output_files.dart
```

Comment:
I'm probably missing a step, but I tried running this command and got this error and it deleted all the generated files

```
[SEVERE] over_react:_over_react_local_builder on test/tmp_test_files/dart2_only/missing_over_react_g_part/library.dart:
Missing "part 'library.over_react.g.dart';".
[SEVERE] over_react:_over_react_local_builder on test/tmp_test_files/mixin_based/missing_over_react_g_part/library.dart:
Missing "part 'library.over_react.g.dart';".
[INFO] Running build completed, took 371ms
[INFO] Caching finalized dependency graph completed, took 351ms
[SEVERE] Failed after 771ms
Build failed; this is expected for "missing_over_react_g_part" files.
Copying built files to golds directory test_fixtures/gold_output_files
Cleaning up temporary directory test/tmp_test_files
Done!
```

Reply:
Hmm those errors are expected since those files intentionally have missing parts, but I wouldn't expect the generated files to be deleted...

Reply:
hmm I tried using this on my other branch and I'm still just getting a bunch of generated files deleted

Reply:
ohhh let me try that

Reply:
okay I see that makes sense! i'm still seeing a few files being added and one more file diff when I run the script than what is checked in for this branch though - [here's what I'm getting](https://github.com/Workiva/over_react/compare/gold-file-workflow-improvements...test-gold-file-changes?expand=1) - I tried pub cleaning but still got the same thing

Reply:
Yeah sorry, I noticed that extra files are being generated, but decided to just leave em there. 

From the PR description:
> It looks like this currently generates some extraneous files, since apparently not all source files are actually used in tests. We can look into this into a follow-up PR

I should really just address that in this PR though; sorry about that.

Reply:
Oh and that one diff I thought I checked in, but forgot to; I'll update that too.

Reply:
Alright those should all be updated, and I added some documentation and the output of that script to talk about what to do if generated files get deleted.